### DISCUSSION TITLE
Request support for RFC 9380 (Hashing to Elliptic Curves)

### DISCUSSION THREAD

**Discussion by Razumain** [bcgit/bc-java#1700] [Ideas]
Is there any plans to support RFC 9380 for hashing to elliptic curves?

Hashing to elliptic curves is a central function necessary for standards like OPRF (RFC 9497) as well as the winning IETF protocol for Password Authenticated Key Exchange (OPAQUE), currently being standardised.

RFC 9380 is necessary to safely hash an arbitrary value to a point or a scalar on an elliptic curve and its general applicability is substantial also outside of the protocols above.

I have a working implementation of RFC 9380 in Java that I could be willing to donate as input. (Contact me if you are interested). It supports Wirestrass curves but not yet Montgomery curves.

**Comment by adamradocz** [comment]
I'd be grateful if I could use the OPAQUE protocol. Any plans to implement it in the near future?

**Comment by msgilligan** [comment]
It looks like this was submitted as a PR and then merged:

* https://github.com/bcgit/bc-java/pull/2277

Squashed commit:

* https://github.com/bcgit/bc-java/commit/dc65548aa73513d9048b16b81366f5ffe59d6255
